paasta_tools/paasta_maintenance.py

```diff
@@ -104,10 +114,71 @@ def is_safe_to_drain(hostname):
     return not are_local_tasks_in_danger()
 
 
-def are_local_tasks_in_danger():
+def is_healthy_in_synapse(local_port, backends):
```

Comment:
Done. We still use the word synapse a lot here. I'm not really sure where we should stop calling it synapse and instead just call it haproxy.